pyp2rpm/convertor.py

```diff
@@ -281,43 +283,57 @@ def metadata_extractor(self):
 
     @property
     def client(self):
-        """XMLRPC client for PyPI. Always returns the same instance.
+        """JSON client for PyPI. Always returns the same instance.
 
         If the package is provided as a path to compressed source file,
         PyPI will not be used and the client will not be instantiated.
 
         Returns:
-            XMLRPC client for PyPI or None.
+            JSON client for PyPI or None.
         """
-        if self.proxy:
-            proxyhandler = urllib.ProxyHandler({"http": self.proxy})
-            opener = urllib.build_opener(proxyhandler)
-            urllib.install_opener(opener)
-            transport = ProxyTransport()
-        if not hasattr(self, '_client'):
-            transport = None
```

Comment:
Proxy handling was moved to the PyPIClient class.  This should fix proxy support that may be currently broken because transport is set to None immediately after constructing it.